qiskit_nature/operators/second_quantization/vibrational_op.py

```diff
@@ -0,0 +1,356 @@
+
+    def adjoint(self) -> "VibrationalOp":
+        dagger_map = {"+": "-", "-": "+", "I": "I"}
+        label_list = []
+        coeff_list = []
+        for label, coeff in zip(self._labels, self._coeffs.tolist()):
+            conjugated_coeff = coeff.conjugate()
+
+            daggered_label = []
+            count = label.count("+") + label.count("-")
+            for char in label:
+                daggered_label.append(dagger_map[char])
+                if char in "+-":
+                    count -= 1
+                    if count % 2 == 1:
+                        conjugated_coeff *= -1
+
+            label_list.append("".join(daggered_label))
+            coeff_list.append(conjugated_coeff)
+
+        return VibrationalOp(list(zip(label_list, np.array(coeff_list, dtype=np.complex128))),
+                             self._num_modes, self._num_modals)
+
+    def reduce(self, atol: Optional[float] = None, rtol: Optional[float] = None) -> "VibrationalOp":
+        if atol is None:
+            atol = self.atol
+        if rtol is None:
+            rtol = self.rtol
+
+        label_list, indices = np.unique(self._labels, return_inverse=True, axis=0)
+        coeff_list = np.zeros(len(self._coeffs), dtype=np.complex128)
+        for i, val in zip(indices, self._coeffs):
+            coeff_list[i] += val
+        non_zero = [
+            i for i, v in enumerate(coeff_list) if not np.isclose(v, 0, atol=atol, rtol=rtol)
+        ]
+        if not non_zero:
+            return VibrationalOp(("I_0*0", 0), self._num_modes, self._num_modals)
+        return VibrationalOp(list(zip(label_list[non_zero].tolist(), coeff_list[non_zero])),
+                             self._num_modes, self._num_modals)
+
+    def compose(self, other: "VibrationalOp") -> "VibrationalOp":
+        if isinstance(other, VibrationalOp):
+            # Initialize new operator_list for the returned Vibration operator
+            new_data = []
+
+            # Compute the product (Vibration type operators consist of a sum of
+            # VibrationalOperator): F1 * F2 = (B1 + B2 + ...) * (C1 + C2 + ...) where Bi and Ci
+            # are VibrationalOperators
+            for label1, cf1 in self.to_list():
+                for label2, cf2 in other.to_list():
+                    new_label, sign = self._single_mul(label1, label2)
+                    if sign == 0:
+                        continue
+                    # TODO: do we need to consider the sign in this application?
+                    new_data.append((new_label, cf1 * cf2 * sign))
+
+            if not new_data:
+                return VibrationalOp(("I_0*0", 0), self._num_modes, self._num_modals)
+
+            return VibrationalOp(new_data, self._num_modes, self._num_modals)
+
+        raise TypeError(
+            f"Unsupported operand type(s) for *: 'VibrationalOp' and '{type(other).__name__}'"
+        )
+
+    # Map the products of two operators on a single fermionic mode to their result.
```

Comment:
Just question. VibrationOp has a algebra same with fermion? or boson?

Reply:
Thanks for spotting this! I did copy a lot of code from `FermionicOp` but the modals are `bosonic` in nature! (confirmed with @paulineollitrault)

I just updated the code to not do the anti-commutation (i.e. take the sign of permutations into account) as we do in the `FermionicOp`. I hope this answers your question :slightly_smiling_face: